### PR TITLE
feat(St-21): added auth based state management

### PR DESCRIPTION
**Pull request by tareq-mahmood-sazim**
## Description of Change

- Refactored according to conventional structure of frontend
- Replaced mantine validations with zod validations
- Added auth based state management using reducers
- Added types
- Added rtk queries for successful authentication
- Added navbar to preview state changing

### Please list which modules are affected by your changes

@/modules/** ,@/redux/reducers/*, @redux/rtk-apis/users, @typedefs/*

---

- [x] **You assure that you have tested the modules properly and there are no breaking changes**
- [x] **You adhere to Sazim's [Engineering Principles](https://docs.google.com/document/d/10mowIwsuYMyNkXS7aeL7AJyVNMYVh1vdMf7yn-A6I1k/edit#heading=h.qqetrc8w2lm8)**

## Associated Ticket Link(s)

- https://trello.com/c/VOo3O01Z

## Related Pull Request(s)

-

### CODE REVIEW

File: shared/typedefs/interfaces.ts
Comment (ishmamur-rahman-sazim) [inline shared/typedefs/interfaces.ts:1]: As per conventions, interface names are generally prefixed with `I`

File: shared/redux/rtk-apis/profiles/profiles.types.ts
Comment (ishmamur-rahman-sazim) [inline shared/redux/rtk-apis/profiles/profiles.types.ts:1]: We can remove this empty file.

File: shared/components/Navbar/components/profileDropdown/profileDropdownButton.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import { useLogout } from "@/shared/hooks/useLogout";
+import { useLazyProfileQuery } from "@/shared/redux/rtk-apis/profiles/profiles.api";
+import { TRootState } from "@/shared/redux/store";
+
+const ProfileDropdownMenu = () => {
+  const [getProfile, { data: profile, isFetching, isError }] = useLazyProfileQuery();
+  const email = useSelector((state: TRootState) => state.authenticatedUser.email);
+  const userId = useSelector((state: TRootState) => state.authenticatedUser.userId);
+  const claim = useSelector((state: TRootState) => state.authenticatedUser.claim);
+
```
Comment (ishmamur-rahman-sazim) [inline shared/components/Navbar/components/profileDropdown/profileDropdownButton.tsx:15]: Can we destructure the preceding lines? 
`const { email, userId, claim } = useSelector((state: TRootState) => state.authenticatedUser);`

File: shared/components/Navbar/components/createClass/components/createClassForm.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+          {...form.getInputProps("classTime")}
+          classNames={{ label: "text-green-500" }}
+          required
+        />
+        <MultiSelect
+          mt="md"
+          label="Days of the Week"
+          placeholder="Select days"
+          classNames={{ label: "text-green-500" }}
+          data={[
```
Comment (ishmamur-rahman-sazim) [inline shared/components/Navbar/components/createClass/components/createClassForm.tsx:53]: [NITPICK] We can have this as an imported constant.

**Review by ishmamur-rahman-sazim — CHANGES_REQUESTED** [review]

File: shared/typedefs/interfaces.ts
Comment (tareq-mahmood-sazim) [reply shared/typedefs/interfaces.ts:1]: All changes have been brought on #3

**Review by ishmamur-rahman-sazim — APPROVED** [review]

**Pull request merged**